scipy/constants/codata.py

```diff
@@ -1333,33 +1705,26 @@ def find(sub=None, disp=False):
     else:
         return result
 
-
-# Table is lacking some digits for exact values: calculate from definition
-c = value('speed of light in vacuum')
-mu0 = 4e-7 * pi
-epsilon0 = 1 / (mu0 * c * c)
-
-exact_values = {
-    'mag. constant': (mu0, 'N A^-2', 0.0),
-    'electric constant': (epsilon0, 'F m^-1', 0.0),
-    'characteristic impedance of vacuum': (sqrt(mu0 / epsilon0), 'ohm', 0.0),
-    'atomic unit of permittivity': (4 * epsilon0 * pi, 'F m^-1', 0.0),
-    'joule-kilogram relationship': (1 / (c * c), 'kg', 0.0),
-    'kilogram-joule relationship': (c * c, 'J', 0.0),
-    'hertz-inverse meter relationship': (1 / c, 'm^-1', 0.0)
-}
-
-# sanity check
-for key in exact_values:
-    val = _current_constants[key][0]
-    if abs(exact_values[key][0] - val) / val > 1e-9:
-        raise ValueError("Constants.codata: exact values too far off.")
-
-physical_constants.update(exact_values)
+exact_values = dict()
+for physical_constant in physical_constants:
+    if physical_constants[physical_constant][2] == 0.0:
+         exact_values[physical_constant] = physical_constants[physical_constant]
+physical_constants.update(exact_values)        
+
+_tested_keys = ['natural unit of velocity',
+                'natural unit of action',
+                'natural unit of action in eV s',
+                'natural unit of mass',
+                'natural unit of energy',
+                'natural unit of energy in MeV',
+                'natural unit of mom.um',
+                'natural unit of mom.um in MeV/c',
+                'natural unit of length',
+                'natural unit of time']
 
 # finally, insert aliases for values
 for k, v in list(_aliases.items()):
-    if v in _current_constants:
+    if v in _current_constants or _tested_keys:
```

Comment:
This is evaluated as `(v in _current_constants) or _tested_keys` which is always truthy. 

```suggestion
    if v in _current_constants or v in _tested_keys:
```

Reply:
This was a step in the right direction but one issue remains: 
The alias substitution applies to the `physical_constants` whereas the `test_find` function searches the `_current_constants` that are unaffected by the alias mechanism. So how should we handle this issue?

Reply:
In this particular case, I'll tell what you tell me to do. I just don't want to be the guy that breaks backwards compatibility.

However, as cluttered this pull request already is, changing the aliases as well their replacement mechanism and adapting the test functions won't do any further harm.

Reply:
git blame shows that `find` was intentionally changed to use `_current_constants` by @pv in 1c274c25c08e31dc008dd698f2ab1f24b71c6cc0.